src/contracts/pods/EigenPodManager.sol

```diff
@@ -134,68 +184,142 @@ contract EigenPodManager is Initializable, OwnableUpgradeable, Pausable, IEigenP
 
-    /**
-     * @notice This function is called to decrement withdrawableRestakedExecutionLayerGwei when a validator queues a withdrawal.
-     * @param amountWei is the amount of ETH in wei to decrement withdrawableRestakedExecutionLayerGwei by
-     */
-    function decrementWithdrawableRestakedExecutionLayerGwei(address podOwner, uint256 amountWei) 
-        external onlyStrategyManager onlyWhenNotPaused(PAUSED_WITHDRAW_RESTAKED_ETH) 
+        _removeShares(slashedPodOwner, shareAmount);
+        _withdrawRestakedBeaconChainETH(slashedPodOwner, slashedFundsRecipient, shareAmount);
+
+        uint256[] memory shareAmounts = new uint256[](1);
+        shareAmounts[0] = shareAmount;
+        IStrategy[] public strategies = new IStrategy[](1);
+        strategies[0] = beaconChainETHStrategy;
+        delegationManager.decreaseDelegatedShares(slashedPodOwner, strategies, shareAmounts);
+    }
+
+    function slashQueuedWithdrawal(
+        address slashedFundsRecipient,
+        BeaconChainQueuedWithdrawal memory queuedWithdrawal
+    )
+        external 
+        onlyOwner 
+        onlyFrozen(queuedWithdrawal.delegatedAddress) 
+        nonReentrant 
     {
-        ownerToPod[podOwner].decrementWithdrawableRestakedExecutionLayerGwei(amountWei);
+        // find the withdrawalRoot
+        bytes32 withdrawalRoot = _calculateWithdrawalRoot(queuedWithdrawal);
+
+         // verify that the queued withdrawal is pending
+        require(
+            withdrawalRootPending[withdrawalRoot],
+            "EigenPodManager.slashQueuedWithdrawal: withdrawal is not pending"
+        );
+
+        // reset the storage slot in mapping of queued withdrawals
+        withdrawalRootPending[withdrawalRoot] = false;
+
+        _withdrawRestakedBeaconChainETH(queuedWithdrawal.podOwner, slashedFundsRecipient, queuedWithdrawal.shares);
     }
 
     /**
-     * @notice This function is called to increment withdrawableRestakedExecutionLayerGwei when a validator's withdrawal is completed.
-     * @param amountWei is the amount of ETH in wei to increment withdrawableRestakedExecutionLayerGwei by
+     * @notice forces a withdrawal of the podOwner's beaconChainETHStrategy shares
+     * @param podOwner is the pod owner whose shares are to be removed
      */
-    function incrementWithdrawableRestakedExecutionLayerGwei(address podOwner, uint256 amountWei) 
-        external onlyStrategyManager onlyWhenNotPaused(PAUSED_WITHDRAW_RESTAKED_ETH) 
+    function forceWithdrawal(address podOwner) external 
+        onlyDelegationManager 
+        onlyWhenNotPaused(PAUSED_WITHDRAWALS)
+        onlyNotFrozen(podOwner)
+        nonReentrant
+        returns (bytes32)
     {
-        ownerToPod[podOwner].incrementWithdrawableRestakedExecutionLayerGwei(amountWei);
+        return _queueWithdrawal(podOwner, podOwnerShares[podOwner], true, true);
+    }
+
+    function getBeaconChainETHShares(address podOwner) external returns (uint256) {
+        return podOwnerShares[podOwner];
```

Comment:
this should be marked 'view'.
It's also super unclear to me why you have a public mapping and a differently-named getter function for the public mapping.